vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/main/java/com/vaadin/flow/component/spreadsheet/CellValueManager.java

```diff
@@ -91,7 +91,7 @@ public class CellValueManager implements Serializable {
     private CellValueHandler customCellValueHandler;
     private CellDeletionHandler customCellDeletionHandler;
 
-    private DataFormatter formatter;
+    private DataFormatter formatter = new CustomDataFormatter();
```

Comment:
Is it needed to create a new instance here?
I see that `updateLocale` creates a new instance of `CustomDataFormatter` using the current locale.

Reply:
It is needed, by my opinion, to keep backwards compatibility: `new CustomDataFormatter()` was called in the constructor in the previous version when `UI.getCurrent() == null`

Newly, we call  `updateLocale` only when `UI.getCurrent() != null`